pkg/segment/results/segresults/segresults.go

```diff
@@ -221,6 +221,12 @@ func (sr *SearchResults) AddSSTMap(sstMap map[string]*structs.SegStats, skEnc ui
 	sr.updateLock.Unlock()
 }
 
+func (sr *SearchResults) AddResultCount(count uint64) {
```

Comment:
There's already `AddBlockResults()` that does this and also merges the block results with the existing results. Shouldn't that function be called? And then I think you don't need to add this function?

Reply:
This is used for groupByCmd and not segStatsCmd, we do not need block results in this case. We just need the initial count.